rfcs/20210510-sum-with-zero-point/README.md

```diff
@@ -0,0 +1,158 @@
+                            "could not append a sum post-op");
+        }
+        ```
+    - Pros:
+        - It is backwards compatible.
+        - Allows to use zero point without scale (scale == 1.0).
+    - Cons:
+        - It is not the simplest solution
+    - Example:
+        ```cpp
+        // Zero point is the last parameter to keep existing code working.
+        attr.append_sum(scale, dt);
+        attr.append_sum(scale, dt, zp);
+        ```
+
+2. Extend existing `append_sum(scale, dt)` and `dnnl_post_ops_append_sum_v2` to
+support zero point.
+    - API:
+        ```cpp
+        // include/oneapi/dnnl/dnnl.h
+        dnnl_status_t DNNL_API dnnl_post_ops_append_sum_v2(
+                dnnl_post_ops_t post_ops, float scale, int32_t zero_point,
+                dnnl_data_type_t data_type);
+
+        // include/oneapi/dnnl/dnnl.hpp
+            void append_sum(float scale = 1.f, int32_t zero_point = 0,
+                    memory::data_type data_type = memory::data_type::undef) {
+                if (data_type == memory::data_type::undef)
+                    error::wrap_c_api(dnnl_post_ops_append_sum(get(), scale),
+                            "could not append a sum post-op");
+                else
+                    error::wrap_c_api(dnnl_post_ops_append_sum_v2(
+                                              get(), scale, zero_point,
+                                              memory::convert_to_c(data_type)),
+                            "could not append a sum post-op");
+            }
+        ```
+    - Pros:
+        - Very simple solution.
+        - Universal solution.
+    - Cons:
+        - It is not backwards compatible for C API.
+    - Example:
+        ```cpp
+        // Zero point is the last parameter to keep existing code working.
+        attr.append_sum(scale, dt);
+        ```
+
+3. Add possibility to use `dnnl_primitive_attr_set_zero_points` with sum.
+    - Pros:
+        - It will allow to use mask and array of zero-point values (instead of
+          one scalar value).
+    - Cons:
+        - More complicated solution.
+        - Probably requires backward incompatibility or adding a new function.
+        - Probably not needed in real topologies.
+    - Example:
+        ```cpp
+        attr.set_zero_points(DNNL_ARG_SRC_1, mask, zero_points);
+        ```
+
+### Sum zero-point design
+
+1. Use two parameters - `arg_zp` and `zero_point` -
+  `append_sum(scale, arg_zp, zp, dt)`:
+    - Pros:
+        - Compatible with scale in append_sum (one scalar value).
+        - Zero-point in sum is currently only needed for src (previous dst) but
+          allows to add an operation for other arguments later. It is also more
+          intuitive than making assumptions for arg.
+    - Cons:
+        - Inconsistent with `set_zero_points` function.
+        - `arg_zp` may be confused with the argument for scale.
+
+2. Use only `zero_point` parameter - `append_sum(scale, zp, dt)` (preferred):
```

Comment:
This seems to be the 1.1 option from previous section but with `zp` before the `dt` parameter). Because it would break the API (a user code with `append_sum(scale, dt)` would fail to compiler), I would recommend to not use this one, but the option in 1.1:
- keep existing order of parameters, 
- add a new C API entry point `dnnl_post_ops_append_sum_v3` with zp argument last.
- and add the new `zp` argument at the end, for the C++ API overload as well

Reply:
Ok, removed. 1.1 option is probably exactly what You described.